stacks-signer/src/runloop.rs

```diff
@@ -145,22 +148,79 @@ pub struct RunLoop<C> {
 }
 
 impl<C: Coordinator> RunLoop<C> {
-    /// Initialize the signer, reading the stacker-db state and setting the aggregate public key
+    /// Initialize the signer runloop
     fn initialize(&mut self) -> Result<(), ClientError> {
-        // Check if the aggregate key is set in the pox contract
-        if let Some(key) = self.stacks_client.get_aggregate_public_key()? {
-            debug!("Aggregate public key is set: {:?}", key);
+        let Some(index) = self.stacks_client.get_signer_index()? else {
+            warn!("Signer index not found. Have you registered your signing key yet?");
+            return Err(ClientError::NotRegistered);
+        };
+        info!("Signer index found. Initializing signer {index}...");
+        self.signing_round.signer_id = index as u32;
+        // TODO: REMOVE THIS AWFULNESS! HIDE THE SHAME! DO NOT LOOK AT ME....
+        // Just a temporary placeholder until we have proper config retrieval from the node.
+        // Then for the love of all that is holy, rip this out and burn it with fire.
+        let signers = self.stacks_client.get_signers().unwrap_or(vec![]);
```

Comment:
This should retry, not fail with `unwrap_or(vec![])`.

Reply:
get_signers already occurs on a retry.

Reply:
EDIT: just saw your other comment. I see what you mean. The deserialization should also be done on a retry not just the call to the node itself.